zp-relayer/workers/poolTxWorker.ts

```diff
@@ -54,14 +62,24 @@ export async function createPoolTxWorker<T extends EstimationType>(gasPrice: Gas
         chainId: CHAIN_ID,
       }
       try {
-        const txHash = await signAndSend(
-          {
-            ...txConfig,
-            ...gasPriceOptions,
-          },
-          config.relayerPrivateKey,
-          web3
-        )
+        const gasPriceValue = await gasPrice.fetchOnce()
+        const gasPriceWithExtra = addExtraGasPrice(gasPriceValue, config.gasPriceSurplus)
+        let txHash
+        try {
+          txHash = await signAndSend(
+            {
+              ...txConfig,
+              ...gasPriceWithExtra,
+            },
+            config.relayerPrivateKey,
+            web3
+          )
+        } catch (e) {
+          // We need to restore nonce to previous value
+          // if transaction wasn't sent
+          await decrNonce()
```

Comment:
Wdyt of the following approach instead?
1. Read nonce from redis during initialisation into memory
2. Use nonce from memory variable when signing transaction
3. Send transaction
4. Increment nonce in memory
5. Update it in redis

Reply:
Tried this approach, but in this case we will always have 1 more redis interaction, so made a little optimisation. If it is not critical, we can stick to the approach you have suggested.

Reply:
Why? We only have one increment operation once per transaction. There is no need to read nonce from redis each time during transaction processing, we could keep in locally in the variable.

Reply:
Ah, I see, will update